src/vivarium/framework/resource/group.py

```diff
@@ -22,42 +25,70 @@ class ResourceGroup:
     def __init__(
         self,
         produced_resources: Iterable[Resource],
-        producer: Any,
-        dependencies: Iterable[Resource] = (),
+        dependencies: Iterable[Resource],
+        initializer: Callable[[SimulantData], None] | None = None,
     ):
+        """Create a new resource group.
+
+        Parameters
+        ----------
+        produced_resources
+            The resources produced by this resource group's producer.
+        dependencies
+            The resources this resource group's producer depends on.
+        initializer
+            The method that initializes this group of resources. If this is
+            None, the resources don't need to be initialized.
+
+        Raises
+        ------
+        ResourceError
+            If the resource group is not well-formed.
+        """
         if not produced_resources:
             raise ResourceError("Resource groups must have at least one resource.")
 
         if len(set(r.resource_type for r in produced_resources)) != 1:
             raise ResourceError("All produced resources must be of the same type.")
 
-        self._resources = list(produced_resources)
-        """The resources produced by this resource group's producer."""
-        self._producer = producer
-        """The method or object that produces this group of resources."""
+        if list(produced_resources)[0].is_initialized != (initializer is not None):
+            raise ResourceError(
+                "Resource groups with an initializer must have initialized resources."
+            )
+
+        self._type = list(produced_resources)[0].resource_type
+        self._resources = {resource.resource_id: resource for resource in produced_resources}
+        self._initializer = initializer
         self._dependencies = dependencies
-        """The resources this resource group's producer depends on."""
 
     @property
     def type(self) -> str:
         """The type of resource produced by this resource group's producer."""
-        return self._resources[0].resource_type
+        return self._type
 
     @property
     def names(self) -> list[str]:
         """The long names (including type) of all resources in this group."""
-        return [resource.resource_id for resource in self._resources]
+        return list(self._resources.keys())
 
     @property
-    def producer(self) -> Any:
-        """The method or object that produces this group of resources."""
-        return self._producer
+    def initializer(self) -> Callable[[SimulantData], None]:
```

Comment:
Is the name change just to get away from overloading the term?

Reply:
I thought it was more consistent terminology. We already talk about simulant initializers elsewhere, and that's what these are.